### ISSUE TITLE
Interpolated Variables & Brief

### ISSUE THREAD

**Issue by devmanhinton** [Amicus/end-dash#73]
https://github.com/amicus/end-dash/tree/interpolated-variables

Allows for

``` html
<div>#{firstName}</div>
```

Instead of

``` html
<div class='firstName-'></div>
```

Issue: Make this work with brief.

A solution:

a) Use a regexp in templates/ ~= 'If a dom element has a class ending in a '-' and no children elements. Add a '#{name}' child text node. Where name is the name of the class ending in a dash.'

b) This leaves affected dom elements with an extra scope reaction (class ending in a dash). You can remove this class but it will break shit tons of css. Solutions:
  I) Add _noOpt as a reserved class name that prevents EndDash scoping reactions
  II) Person up, and break all the css, then fix all the css

Thoughts? @yaymukund @bglusman

**Comment by tobowers** [comment]
I'd at least go to data-no-op-enddash or something so it's not a very strange class name in the HTML. The syntax is much better though as long as it doesn't break server-side pre-rendering.

Question is if you can do that and fix brief in the remainder of this sprint.

**Comment by devmanhinton** [comment]
Not sure. Will have to think on that.

Does fixing brief count as bug fixing ? :D

**Comment by yaymukund** [comment]
Instead of adding a special no-op flag, how about we support the old style (call it deprecated, don't document it) and then get brief to use the new style piece-by-piece?

**Comment by tobowers** [comment]
that'd be much better and @devmanhinton had talked about looking for "no text nodes"

**Comment by bglusman** [comment]
Yeah, Devon & I discussed this but it'd be hard since this branch gets rid of rules.js entirely... another possibility was bumping the minor version for this change and leaving brief on non-master, able to bump patch version for bug fixes, and make a new brief branch we work on moving over to new syntax using master which we merge to brief master when confident/passes rain-forest tests, etc.

**Comment by devmanhinton** [comment]
@tobowers I did mention that but adding that check  ('if there is a scope reaction on a dom element but that dom element has no non-text node children) would prevent the following, valid behavior:

Model:

``` js
Model {person: new Backbone.Model({firstName: 'Derrick'})
```

HTML:

``` html
<div class='person-'>#{firstName}</div>
```

**Comment by yaymukund** [comment]
Removing rules.js seems like a big architectural change. Do you think it makes sense to do that in its own branch?

**Comment by devmanhinton** [comment]
Yes.

**Comment by tobowers** [comment]
Why does that prevent it?

``` html
<div class='person-'>#{firstName}</div>
```

would currently be written as:

``` html
<div class="person-"><span class="firstName-"></span></div>
```

And in that case the catch for "not doing scoping" would apply to firstName- but not person- in both cases, right?

**Comment by devmanhinton** [comment]
Yeah that is true.
